demo/src/main/java/Y/Player.java

```diff
@@ -0,0 +1,107 @@
+package Y;
+
+import java.util.*;
+
+public class Player {
+    private String name;
+    private int bucket;
+    private int bucketSize = 10;
+    private int milkingSkill = 1;
+    private List<Milkables> ownedAnimals = new ArrayList<Milkables>();
+
+    public Player(String name) {
+        this.name = name;
+    }
+
+    public List<Milkables> getAnimalList() {
+        return this.ownedAnimals;
+    }
+    
+    public int getAnimalsLen() {
+        return ownedAnimals.size();
+    }
+    
+    public Milkables getAnimal(int animalIndex) {
+        return this.ownedAnimals.get(animalIndex);
+    }
+    
+    public void setBucket(int i) {
+        this.bucket = i;
+    }
+
+    public int getBucket() {
+        return this.bucket;
+    }
+
+    public int getBucketsize() {
+        return this.bucketSize;
+    }
+
+    public void setBucketsize(int i) {
+        this.bucketSize = i;
+    }
+
+    public void upgradeBucketSize() {
+        if (getBucket() == getBucketsize() && (getBucketsize() * 2) < 160000) {
+                setBucketsize(getBucketsize() * 2);
+                System.out.println("\nBucket size increased to " + getBucketsize());
+                setBucket(0);
+        } else {
+            System.out.println("\nYou can't afford that or the bucket is fully upgraded.");
+        }
+    }
+
+    public int getMilkingSkill() {
+        return this.milkingSkill;
```

Comment:
No need to reference `this`. It's only used if there are other variables with the same name as an instance variable, then `this.sameVariable` works to keep them apart, e.g. in setters.